rpc/rpcserver/server.go

```diff
@@ -2346,6 +2346,189 @@ func (s *loaderServer) FetchMissingCFilters(ctx context.Context, req *pb.FetchMi
 	return &pb.FetchMissingCFiltersResponse{}, nil
 }
 
+func (s *loaderServer) RpcSync(req *pb.RpcSyncRequest, svr pb.WalletLoaderService_RpcSyncServer) error {
+	defer zero.Bytes(req.Password)
+
+	// XXX: Do it need to lock here like StartConsensusRpc
+	defer s.mu.Unlock()
+	s.mu.Lock()
+
+	if s.rpcClient != nil {
+		return status.Errorf(codes.FailedPrecondition, "RPC client already created")
+	}
+
+	networkAddress, err := cfgutil.NormalizeAddress(req.NetworkAddress,
+		s.activeNet.JSONRPCClientPort)
+	if err != nil {
+		return status.Errorf(codes.InvalidArgument, "Network address is ill-formed: %v", err)
+	}
+
+	// Error if the wallet is already syncing with the network.
+	wallet, walletLoaded := s.loader.LoadedWallet()
+	if walletLoaded {
+		_, err := wallet.NetworkBackend()
+		if err == nil {
+			return status.Errorf(codes.FailedPrecondition, "wallet is loaded and already synchronizing")
+		}
+	}
+
+	if req.DiscoverAccounts && len(req.PrivatePassphrase) == 0 {
+		return status.Errorf(codes.InvalidArgument, "private passphrase is required for discovering accounts")
+	}
+	var lockWallet func()
+	if req.DiscoverAccounts {
+		lock := make(chan time.Time, 1)
+		lockWallet = func() {
+			lock <- time.Time{}
+			zero.Bytes(req.PrivatePassphrase)
```

Comment:
Don't know if it's a problem, so just to confirm.

Only doing the clearing here means `req.PrivatePrassphrase` is *not* zeroed if the `wallet.Unlock()` call below fails which is slightly different than how this is performed in other functions in rpcserver (where the zeroing is a defered call  defined at the top of the request and unconditionally executed).

Reply:
Hrm yah I suppose that's true, I can move up the defer so it's caught too in case of Unlock fails.